VTUL/dlp-access
# LIBTD-2347: Allow SiteAdmin users to upload to html/ and images/ subf…

…olders in the site content S3 bucket

**JIRA Ticket**: (https://webapps.es.vt.edu/jira/browse/LIBTD-2347) (:star:)

# What does this Pull Request do? (:star:)
This PR sets up a site content upload page allowing an siteAdmin user to upload site content files and store them in S3 bucket. Then the site will get the corresponding site content files either from browser session storage or S3 storage.

# What's the changes? (:star:)

* cypress/fixtures and cypress/integration/admin_page_upload_file.spec.js: adds tests for uploading files to S3 bucket as well as files in cypress fixture
* ContentUpload.js: site content upload page
* lib/fetchTools.js: gets file (text or image) first from session, if not existed, get it from S3 storage
* AboutPage.js, PermissionPage.js, AdditionalPage.js: get html file from S3 storage
* CollectionHighlights.js, FeaturedItem.js, SiteSponsors.js, FeaturedStaticImage.js: get image file from S3 storage

# How should this be tested?

* Go to siteAdmin/, and then contentUpload page to upload a site content file (refer to 
VTDLP-sitecontent)
* Clear session cache and check if the changed content/image is reflected on the site page

# Interested parties
Tag (@yinlinchen) interested parties

(:star:) Required fields
